source/postprocess/gravity_point_values.cc

```diff
@@ -0,0 +1,322 @@
+         << minimum_longitude << ' ' << maximum_longitude << '\n'
+         << minimum_latitude << ' ' << maximum_latitude << '\n';
+
+      // Storing cartesian coordinate, density and JzW at quadrature points in a vector
+      // avoids to use MaterialModel and fe_values within the loops.
+      unsigned int local_cell_number = (this->get_triangulation().n_locally_owned_active_cells());
+      const unsigned int number_quadrature_points_mpi = local_cell_number * number_quadrature_points_cell;
+      std::vector<double> density_JxW (number_quadrature_points_mpi);
+      std::vector<Point<dim> > position_point (number_quadrature_points_mpi);
+
+      // Store density and from MaterialModel and allocate to density_all array:
+      typename DoFHandler<dim>::active_cell_iterator
+      cell = this->get_dof_handler().begin_active(),
+      endc = this->get_dof_handler().end();
+      MaterialModel::MaterialModelInputs<dim> in(quadrature_formula.size(),this->n_compositional_fields());
+      MaterialModel::MaterialModelOutputs<dim> out(quadrature_formula.size(),this->n_compositional_fields());
+      local_cell_number = 0;
+      for (; cell!=endc; ++cell)
+        {
+          if (cell->is_locally_owned())
+            {
+              fe_values.reinit (cell);
+              const std::vector<Point<dim> > &position_point_cell = fe_values.get_quadrature_points();
+              in.reinit(fe_values, cell, this->introspection(), this->get_solution(), false);
+              this->get_material_model().evaluate(in, out);
+              for (unsigned int q = 0; q < number_quadrature_points_cell; ++q)
+                {
+                  density_JxW[local_cell_number * number_quadrature_points_cell + q] = out.densities[q] * fe_values.JxW(q);
+                  position_point[local_cell_number * number_quadrature_points_cell + q] = position_point_cell[q];
+                }
+              ++local_cell_number;
+            }
+        }
+
+      // loop on r - satellite position [r, ,  ]
+      for (unsigned int h=0; h < number_points_radius; ++h)
+        {
+          std_cxx11::array<double,dim> satellite_coordinate;
+          satellite_coordinate[0] = minimum_radius + ((maximum_radius - minimum_radius) / number_points_radius) * h;
+
+          // loop on phi - satellite position [ , phi ,]
+          for (unsigned int i=0; i < number_points_longitude; ++i)
+            {
+              satellite_coordinate[1] = (minimum_longitude + ((maximum_longitude - minimum_longitude) / number_points_longitude) * i) * numbers::PI / 180.;
+
+              // loop on theta - satllite position [ , , theta]
+              for (unsigned int j=0; j < number_points_latitude; ++j)
+                {
+                  satellite_coordinate[2] = (minimum_latitude + ((maximum_latitude - minimum_latitude) / number_points_latitude) * j) * numbers::PI / 180.;
+                  Tensor<1,dim> local_g;
+                  double local_U = 0;
```

Comment:
What is U? Is it the gravity potential? then name it `local_gravity_potential`. There are too many letters in too many different scientific fields to remember them all (also there is often some ambiguity). I am not so much concerned about the `g` above because that is really universal.